source/common/network/socket_interface_impl.h

```diff
@@ -44,12 +44,17 @@ class SocketInterfaceImpl : public SocketInterfaceBase {
                                                 absl::optional<int> domain,
                                                 Io::IoUringFactory* io_uring_factory = nullptr);
 
+  // TODO (soulxu): making those configurable if needed.
+  static constexpr uint32_t DefaultIoUringSize = 300;
+  static constexpr uint32_t DefaultReadBufferSize = 8192;
+  static constexpr bool UseSubmissionQueuePolling = false;
+
 protected:
   virtual IoHandlePtr makeSocket(int socket_fd, bool socket_v6only, absl::optional<int> domain,
                                  Io::IoUringFactory* io_uring_factory = nullptr) const;
 
 private:
-  std::unique_ptr<Io::IoUringFactory> io_uring_factory_;
+  std::weak_ptr<Io::IoUringFactory> io_uring_factory_;
```

Comment:
Is it ok to use shared ptr here?

Reply:
No, which will cause io_uring_factory_ lives as long as SocketInterfaceImpl (static).